src/Blocks/sources.jl

```diff
@@ -730,3 +731,145 @@ end
 function SampledData(; name, buffer, sample_time, circular_buffer)
     SampledData(SampledDataType.vector_based; name, buffer, sample_time, circular_buffer)
 end
+
+"""
+    Interpolation(interp_type, u, x, args...; name)
+
+Represent function interpolation symbolically as a block component.
+By default interpolation types from [`DataInterpolations.jl`](https://github.com/SciML/DataInterpolations.jl) are supported,
+but in general any callable type that builds the interpolation object via `itp = interpolation_type(u, x, args...)` and calls
+the interpolation with `itp(t)` should work. This does not need to represent an interpolation, it can be any type that satisfies
+the interface, such as lookup tables.
+# Arguments:
+  - `interp_type`: the type of the interpolation. For `DataInterpolations`,
+these would be any of [the available interpolations](https://github.com/SciML/DataInterpolations.jl?tab=readme-ov-file#available-interpolations),
+such as `LinearInterpolation`, `ConstantInterpolation` or `CubicSpline`.
+  - `u`: the data used for interpolation. For `DataInterpolations` this will be an `AbstractVector`
+  - `x`: the values that each data points correspond to, usually the times corresponding to each value in `u`.
+  - `args`: any other arguments beeded to build the interpolation
+# Keyword arguments:
+  - `name`: the name of the component
+
+# Parameters:
+  - `interpolator`: the symbolic
+
+# Connectors:
+  - `input`: a [`RealInput`](@ref) connector corresponding to the independent variable
+  - `output`: a [`RealOutput`](@ref) connector corresponding to the interpolated value
+"""
+function Interpolation(interp_type, u, x, args...; name)
+    itp = interp_type(u, x, args...)
+    Interpolation(itp; name)
+end
+
+function Interpolation(itp; name)
+    @parameters (interpolator::typeof(itp))(..) = itp
+    @named input = RealInput()
+    @named output = RealOutput()
+
+    eqs = [output.u ~ interpolator(input.u)]
+
+    ODESystem(
+        eqs, t, [], [interpolator]; name, systems = [input, output])
+end
+
+struct CachedInterpolation{T, I, U, X, C}
```

Comment:
A docstring would be helpful for future development, even if it's not public API.